synapse/python_dependencies.py

```diff
@@ -83,19 +87,31 @@ def list_requirements():
 
 
 class DependencyException(Exception):
+    @property
+    def message(self):
+        return "\n".join([
+            "Missing Requirements: %s" % (", ".join(self.dependencies),),
+            "To install run:",
+            "    pip install --upgrade --force %s" % (" ".join(self.dependencies),),
+            "",
+        ])
```

Comment:
I'd be tempted to pull this up as a top level template string and then use the multiline quotes, but that's a matter of taste

Reply:
yeah, that would have been a good idea. I just c&ped it tbh.

Reply:
Fair